automation/roles/etcd/tasks/main.yml

```diff
@@ -86,6 +86,14 @@
     state: directory
   tags: etcd, etcd_conf
 
+- name: Fetch etcd TLS certificate, key and CA from the master node
+  ansible.builtin.include_role:
+    name: ../roles/tls_certificate/copy
+  vars:
+    copy_for: "etcd"
+  when: tls_cert_generate|bool
+  tags: etcd, etcd_conf
```

Comment:
I think we could call this role at the `etcd_cluster.yml` playbook level rather than from the tasks file.
```
   - role: tls_certificate/copy
      vars:
        copy_for: "etcd"
```

Reply:
I placed it there so the certificate is fetched immediately after the `/etc/etcd` directory is created. Adding it before the execution of the etcd role won't work because the `/etc/etcd` directory doesn't exist yet. On the other hand, running it after the etcd role will cause the etcd daemon to fail to start due to missing certificates.

Reply:
I placed it there so the certificate is fetched immediately after the `/etc/etcd` directory is created and etcd is installed (but not yet configured). Adding it before the execution of the etcd role likely won't work because etcd won't be installed yet, meaning the etcd user and group will be missing, and changing the file ownership will fail. On the other hand, running it after the etcd role will cause the etcd daemon to fail to start due to missing certificates.

Reply:
Ok, thanks